### PR TITLE
Added the Hypothesis Architecture Doc

### PR DESCRIPTION
**Pull request by mik-dass**
Signed off by : mik-dass <mrinald7@gmail.com>

### CODE REVIEW

File: HypothesisArchitecture.asciidoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+* Expose an experiment management api. This api is the mechanism used to create and coordinate experiments. This coordination includes configuration, execution and lifecycle semantics of isolated experiments. Another management aspect is the user driven design of schemas that serve as contracts for measurement data transmitted to the supervisor. Experiments are the means to collect measurements provided by scouts deployed and distributed throughout the system.  This data collection activity is responsible for validating, cleansing, transforming, filtering, correlating, aggregating and potentially storing ( for non transient ) collected data at predetermined intervals in an isolated manner. Repo link : https://github.com/fabric8-hdd/fabric8-hypothesis-manager-api
+* Expose a data retrieval api to make available experimental data. Repo link : https://github.com/fabric8-hdd/fabric8-hypothesis-data-api
+* Expose a data collection api that serves as the central destination for data sent by observing scouts. Repo link : https://github.com/fabric8-hdd/fabric8-hypothesis-measurements-api
+
```
Comment (anmolbabu) [inline HypothesisArchitecture.asciidoc:35]: We need a way to say that due to the differences in functionality and scaling requirements, each of the 3 different api classifications become candidates for autonomous micro-services and accordingly they have been maintained as part of 3 different repos(then repos being different for each other actually gets a justification)..
I would also be happy to see a diagram that shows this division to make things more clearer...

Comment (mik-dass) [reply HypothesisArchitecture.asciidoc:35]: I guess the diagram is there, right?

File: HypothesisArchitecture.asciidoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Introduction
+
+The hypothesis engine is a service that enables the user to test hypotheses about solutions deployed using Openshift.io. If Openshift can be thought of as an enterpriseenterprize kubernetes environment for cloud native solutions, then Openshift.io can be thought of as a cloud based tool set to plan build and deploy these solutions in that environment.
```
Comment (anmolbabu) [inline HypothesisArchitecture.asciidoc]: please fix the enterprise spelling

File: HypothesisArchitecture.asciidoc (hunk 1 above)
Comment (anmolbabu) [reply HypothesisArchitecture.asciidoc:35]: As we discussed, I was talking about the diagram that shows the responsibilities of supervisor into 3 + 1 micro-services..

Comment (mik-dass) [reply HypothesisArchitecture.asciidoc:35]: Ok But I will ping @mitchharpur for that diagram and add it later

**Review by anmolbabu — APPROVED** [review]

**Pull request merged**